### PR TITLE
Add scan QR code optional CTA to LoginMagicLinkSentImprovedFragment

### PR DESCRIPTION
**Pull request by JorgeMucientes**
## Description 

~~Adds new CTA to scan QR code from magic link sent screen. The goal was to add the least possible changes in the auth lib to ensure that backwards compatibility is not broken in case any other apps are using this screen.~~

~~Most of the changes are [done in this PR](https://github.com/woocommerce/woocommerce-android/pull/7740) inside WooCommerce Android repository. Nothing to test in this PR.~~

**EDIT**

After @hichamboushaba suggestion I removed `LoginMagicLinkSentImprovedFragment` from this library and migrated it to WooCommerce Android repo. So this PR just does that -> remove `LoginMagicLinkSentImprovedFragment`

## Screenshots 

ScanQrCta

### CODE REVIEW

**Comment by hichamboushaba** [comment]
@JorgeMucientes A suggestion, since we are planning on making more changes to this screen, why not just move it completely to Woo? I [suggested ](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/88#issuecomment-1201379669) this to @0nko before, and we decided to skip it because the work was already done, but now I see we are doing more changes, then maybe it's simpler to copy the code of the improved magic link screen and drop it in Woo, then we can get rid of the inheritance which makes code harder to reason about.

**Comment by JorgeMucientes** [comment]
Makes sense @hichamboushaba. We are sure WordPress is not using this right? I'll get to it!

**Comment by hichamboushaba** [comment]
> We are sure WordPress is not using this right?

I looked at the WPAndroid repo now, and I confirm it's not used anywhere.

File: WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginMagicLinkSentImprovedFragment.java [hunk 1]
```diff
@@ -1,151 +0,0 @@
...
-import androidx.appcompat.app.ActionBar;
-import androidx.appcompat.app.AppCompatActivity;
-import androidx.appcompat.widget.Toolbar;
-import androidx.fragment.app.Fragment;
-
-import javax.inject.Inject;
-
-import dagger.android.support.AndroidSupportInjection;
-
-public class LoginMagicLinkSentImprovedFragment extends Fragment {
```
Comment (JorgeMucientes) [inline WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginMagicLinkSentImprovedFragment.java:24]: Migrating this to Kotlin using newer language and tools felt ---> Back to the future [LoginMagicLinkSentImprovedFragment](https://github.com/woocommerce/woocommerce-android/pull/7740/files#diff-9b576a79aed5c1efb573e8698c577749e8df56664b41ef440607948dc6504f71)

**Comment by JorgeMucientes** [comment]
Ready for review @hichamboushaba @0nko 🙏🏼

**Review by 0nko — APPROVED** [review]
:shipit:

**Pull request merged**